### ISSUE TITLE
Announcements Repeat too frequently, not very random

### ISSUE THREAD

**Issue by iBuilderLive** [TNotifier/Issues#17]
when it comes to the announcements, I know they are supposed to be random, but do you think you could add a check to  make sure the announcement does not equal the last announcement?  simple enough check.  I am noticing that announcements are often, very often in fact, repeating.

something like  (and clearly I'm no pro here but you'll get the gist)

integer currA  = currentlyDisplayedAnnouncementID;    //assuming you're using record id#'s
integer nextA = getNextAnnoucement();  //gets a random announcement
checkNextA(currA, nextA);

checkNextA(integer cA, integer nA); {
    if (cA != nA){
         cA = nA;
         return cA;
    } else {
         nextA = getNextAnnoucement();
         checkNextA(currA, nextA);
    }
}

I know this could end in an endless loop if there are only 1 announcement you can add a check # of announcements saved (not knowing how you've saved announcements I'll leave that to you)

or even give us an option to make the announcement play sequentially?

**Comment by Oyed** [comment]
Will add this tonight.

**Comment by Oyed** [comment]
I've gone a step further than making Announcements a little intelligent, now Announcements are fully intelligent in what to display. For example:

Announcement 1
Announcement 2
Announcement 3

Before, it'd just randomly pick one. No confusing logic there. Means if you're unlucky, you could get the same announcement numerous times in a row, which is what you were probably experiencing. Now, it does this:

Show Announcement -> pick random (Announcement 1) -> Take out "Announcement 1" from possible announcements
Show Announcement -> pick random (Announcement 3) -> Take out "Announcement 3" from possible announcements
Show Announcement -> pick random (Announcement 2) -> Take out "Announcement 2" from possible announcements
Show Announcement -> no more left -> restore possible announcements

And then it repeats.

TL;DR Announcements have the same random aspect, but will not play the same one more than once before there has been a full rotation of all announcements.

Thanks!

**Comment by iBuilderLive** [comment]
perfect, awesome as usual bro.  keep up the amazIng work
